### ISSUE TITLE
mAP value decrease when i try to continue trainning with the same dataset

### ISSUE THREAD

**Issue by xiuyi007** [ultralytics/ultralytics#14359]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello! I have trained a model on my own dataset, and got a best.pt. For some reason, i want to use this best.pt to continue training with the same dataset. The problem is the mAP50 value of  first epoch is only 0.856( i think it should be around 0.9) and the next few epochs even worse. Untill epoch16, the mAP value comes to 0.9.
Here is part of code, mTrain.py/mVal.py is used to train or val:
```python
if __name__ == '__main__':

    parser = argparse.ArgumentParser(description='YOLO Version argument')
    parser.add_argument('--v', type=str, default='8', help='yolo version chosed')
    parser.add_argument('--improve', type=str, default=None, help='improved version')
    parser.add_argument('--model', type=str, default=None, help='yolo model')
    parser.add_argument('--epochs', type=int, default=20, help='epoch number')
    # parser.print_help()

    opt = parser.parse_args()

    if opt.model:
        model = YOLO(opt.model)
    else:
        file_name = f"cfg/models/yolov{opt.v}s.yaml" if opt.improve is None else f"cfg/models/yolov{opt.v}s-{opt.improve}.yaml"
        model = YOLO(file_name)
    model.train(data="cfg/datasets/pest.yaml", epochs=opt.epochs, imgsz=640, batch=32)
```
### In the terminal: 
` python mTrain.py --model runs/detect/8s/weights/best.pt --epochs 20`
### results:
![image](https://github.com/ultralytics/ultralytics/assets/82797119/89bdf2b9-61c0-4ba6-818a-a1853f931f72)

So,  I run code below to check the last.pt
```python
if __name__ == '__main__':

    parser = argparse.ArgumentParser(description='YOLO Version argument')
    parser.add_argument('--v', type=str, default='8', help='yolo version chosed')
    parser.add_argument('--improve', type=str, default=None, help='improved version')
    # parser.print_help()

    opt = parser.parse_args()

    file_name = f"runs/detect/{opt.v}s/weights/best.pt" if opt.improve is None else f"runs/detect/v{opt.v}s_{opt.improve}/weights/best.pt"

    model = YOLO(file_name)
    model.val()
```
### In the terminal:
` python mVal.py --v 8`
### results:
![image](https://github.com/ultralytics/ultralytics/assets/82797119/0fe0d189-ae75-498f-a17a-ee6c963defd7)

training results of the best.pt(first training):
![image](https://github.com/ultralytics/ultralytics/assets/82797119/fc191281-46c4-4262-8ec8-78b1fbb7eb11)
so why is the model's performance  not meeting my expectations during the early stage of second training?

**Comment by glenn-jocher** [comment]
@xiuyi007 hello!

Thank you for reaching out and providing detailed information about your issue. It's great to see that you've already done some troubleshooting on your own. Here are a few suggestions to help address the mAP value decrease when continuing training:

1. **Learning Rate Adjustment**: When resuming training, the learning rate might need to be adjusted. The default learning rate might be too high for fine-tuning, causing the model to initially perform worse. You can try lowering the learning rate when resuming training:
    ```python
    model.train(data="cfg/datasets/pest.yaml", epochs=opt.epochs, imgsz=640, batch=32, lr0=0.001)
    ```

2. **Warmup Phases**: Ensure that the warmup phase is appropriately set. Sometimes, a short warmup phase can help stabilize training when resuming:
    ```python
    model.train(data="cfg/datasets/pest.yaml", epochs=opt.epochs, imgsz=640, batch=32, warmup_epochs=3)
    ```

3. **Check for Overfitting**: If the model's performance improves after several epochs, it might be overfitting to the training data. You can monitor the validation loss and consider using early stopping or regularization techniques.

4. **Model State and Optimizer State**: Ensure that the optimizer state is also being restored when resuming training. This can help maintain the momentum and other optimizer-specific parameters:
    ```python
    model.train(data="cfg/datasets/pest.yaml", epochs=opt.epochs, imgsz=640, batch=32, resume=True)
    ```

5. **Reproducible Example**: If the issue persists, providing a minimum reproducible example can significantly help in diagnosing the problem. You can refer to our guide on creating a reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

6. **Latest Versions**: Ensure that you are using the latest versions of the Ultralytics YOLO package and other dependencies. Sometimes, updates include bug fixes and performance improvements that could resolve your issue.

Please try these suggestions and let us know if the issue persists. We're here to help!

**Comment by xiuyi007** [comment]
Acutually, it doesn't work.Could this be due to the data augmentation in YOLO? I mean the datasets obtained after data augmentation are inconsistent between my first training session and my second training session.

**Comment by pderrenger** [comment]
@xiuyi007 it's possible that data augmentation could be causing inconsistencies between training sessions. Data augmentation introduces variability, which might lead to differences in the initial loss values when resuming training. To mitigate this, you can try disabling or reducing augmentation when resuming training to see if it stabilizes the results. Additionally, ensure you're using the latest version of the Ultralytics YOLO package to rule out any bugs that might have been fixed in recent updates. If the issue persists, please provide more details about your augmentation settings.

**Comment by xiuyi007** [comment]
Thanks. May i ask another question?I have a trained v10 model, and I want to train a new v10 model. But I only want to load the weights of the backbone part of this model, and the other weights should be random. How can I do this?

**Comment by pderrenger** [comment]
@xiuyi007 you can achieve this by loading the pretrained weights and then selectively freezing or reinitializing the layers you want to train from scratch. In YOLOv10, you can load the model with pretrained weights and then modify the layers as needed. If you need further assistance, please refer to the Ultralytics documentation or ask for specific code examples.

**Comment by xiuyi007** [comment]
How to reinitialize the layers i want to train from scratch? Can you give me some code demo?

**Comment by pderrenger** [comment]
To reinitialize specific layers while keeping others from a pretrained YOLOv10 model, you can load the model, extract the backbone weights, and then reinitialize the desired layers. Here's a minimal example to get you started:

```python
from ultralytics import YOLO

# Load the pretrained model
model = YOLO('path/to/pretrained_model.pt')

# Reinitialize specific layers
for layer in model.model.backbone.layers_to_reinitialize:
    layer.reset_parameters()

# Continue training
model.train(data='path/to/data.yaml', epochs=100)
```

This example shows how to reset parameters for specific layers. Adjust the `layers_to_reinitialize` to match the layers you want to reinitialize. For more details, refer to the Ultralytics documentation.
